spec/controllers/games_controller_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'rails_helper'
+
+RSpec.describe GamesController, type: :controller do
+
+  describe "games#new action" do 
+    it "should require users to be logged in" do 
+      get :new
+      expect(response).to redirect_to new_user_session_path
+    end
+    
+    it "should return success response on the get request" do
+      user = create(:user)
+      sign_in user
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "games#show action" do 
+    it "should return success response on the get request" do
+      game = create(:game)
+      get :show, id: game.id
+      expect(response).to have_http_status(:success)
+    end
+
+    it "should return a Game Not Found error if the game is not found" do 
+      get :show, id: '99999999'
+      expect(response).to have_http_status(:not_found)
+    end
+  end
+
+  describe "games#update action" do 
+    before :each do 
+      @game = FactoryGirl.create(:game, name: "Test Name", state: "open")
+    end
+
+    context "with valid attributes" do
+      it "located the requested @game" do
+        user = create(:user)
+        sign_in user
+        put :update, id: @game, game: FactoryGirl.attributes_for(:game)
+        expect(@game).to eq(@game) 
```

Comment:
Well really, the main problem is that you're comparing the instance variable `@game` created in the `before` block to itself. In the article that you shared, the writer is using the `assigns` hash (http://guides.rubyonrails.org/testing.html#the-four-hashes-of-the-apocalypse) to compare the instance variable created in the spec to the one created in the controller action. That would work here but isn't compatible with my suggestion to just use a local variable in the action, which is why it seems easiest to simply roll this into the next example.

Another reason for doing so is that testing a specific controller variable would couple the test too closely to the controller implementation. Ultimately, it doesn't matter what intermediate variables are used to accomplish the task as long as the action properly updates the correct game, so the tests should aim to establish this in the least brittle way possible.